src/drivers/intel/ssp/mn.c

```diff
@@ -291,7 +291,7 @@ void mn_release_mclk(uint32_t mclk_id)
 	if (!mn->mclk_sources_ref[mclk_id]) {
 		mdivc = mn_reg_read(MN_MDIVCTRL, mclk_id);
 
-		mdivc |= ~MN_MDIVCTRL_M_DIV_ENABLE(mclk_id);
+		mdivc &= ~MN_MDIVCTRL_M_DIV_ENABLE(mclk_id);
```

Comment:
that looks indeed like a legit fix.

Reply:
nice catch, thanks for the fixing, yes, we need this being back-ported to the impacted stable branches.